util/oidc/oidc.go

```diff
@@ -148,17 +148,21 @@ func NewClientApp(settings *settings.ArgoCDSettings, dexServerAddr string, dexTl
 	return &a, nil
 }
 
-func (a *ClientApp) oauth2Config(scopes []string) (*oauth2.Config, error) {
+func (a *ClientApp) oauth2Config(request *http.Request, scopes []string) (*oauth2.Config, error) {
```

Comment:
It seems like the request object is needed here. Re: https://github.com/argoproj/argo-cd/pull/14208/files#r1638640243